### PR TITLE
cmd/get-authkey: add an OAuth API client to produce an authkey

### PR DESCRIPTION
**Pull request by DentonGentry**
Updates https://github.com/tailscale/tailscale/issues/3243

Signed-off-by: Denton Gentry <dgentry@tailscale.com>

### CODE REVIEW

**Comment by willnorris** [comment]
From a quick glance (without actually running it) LGTM.  Needs godoc though... at the very least at the package level describing the binary.

**Review by willnorris — APPROVED** [review]

File: cmd/get-authkey/main.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			},
+		},
+	}
+
+	authkey, _, err := tsClient.CreateKey(ctx, caps)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	fmt.Println(authkey)
```
[the remark this answers is not in the document]
Comment (DentonGentry) [reply cmd/get-authkey/main.go:65]: Yes, thank you CodeQL.

File: cmd/get-authkey/main.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// get-authkey allocates a one-time-use authkey using an OAuth API client
```
Comment (willnorris) [inline cmd/get-authkey/main.go:4]: Doc says one time use, but it also accepts reusable flag?

Comment (DentonGentry) [reply cmd/get-authkey/main.go]: Done

**Pull request merged**